spec.html

```diff
@@ -38886,6 +38882,22 @@ <h1>WeakRef.prototype [ @@toStringTag ]</h1>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-weakref-abstract-operations">
+      <h1>WeakRef Abstract Operations</h1>
+      <emu-clause id="sec-weakrefderef" aoid="WeakRefDeref">
+        <h1>WeakRefDeref ( _weakRef_ )</h1>
+        <p>The abstract operation WeakRefDeref takes argument _weakRef_ (a WeakRef). It performs the following steps when called:</p>
+        <emu-alg>
+          1. Perform ? RequireInternalSlot(_weakRef_, [[WeakRefTarget]]).
+          1. Let _target_ be _weakRef_.[[WeakRefTarget]].
+          1. If _target_ is not ~empty~, then
+            1. Perform ! AddToKeptObjects(_target_).
+            1. Return _target_.
+          1. Return *undefined*.
+        </emu-alg>
+      </emu-clause>
```

Comment:
It's probably worth adding a `NOTE` to this clause explaining that this is a distinct operation strictly for making it easier to define liveness.